Pocket/pocket-ios
# feat(tags): refine recent tags experience

## Summary
Refining recent tags experience: 
* Allow only Premium users to view the recent tags and if their tags are greater than 3
* Updating UI and Unit Tests
* Adding Analytics for tapping on recent tags

## References 
IN-1268

## Implementation Details
* Added a check to `recentTags` to be displayed if user is Premium and if the number of all their tags is greater than 3.
* Added two tracking called `addTagsRecentTagTapped` and `filterTagsRecentTagTapped`
* Added `user` in `Services` for shared extension
* Moved `RecentTag` to its own file
* Created test for recent tags

## Test Steps
Please 
There should be two sections: (1) Recent Tags, which shows the three most-recently used tags, and (2) All Tags, which shows all of a user’s tags (including the three most-recently used). Both sections should be sorted alphabetically. 
1. Recent Tags and Add Tags screens should be consistent 
2. Recent Tags should appear first in the tags list 
3. Recent Tags should be distinguishable in some way 
4. Recent Tags should be sorted by most-recent use
5. All Tags should be sorted by alphabetical order
6. Recent Tags should only be visible to Premium Users
7. Add x to right side of tag for Add Tags screen

## PR Checklist:

## Screenshots

https://user-images.githubusercontent.com/6743397/231266937-728ce0ce-3bbe-4236-b5b1-76abfd368a1b.mp4
